frontend/src/router.ts

```diff
@@ -233,24 +201,24 @@ export const checkAuth: NavigationGuard = async (to, from, next) => {
   const currentUser = firebase.auth().currentUser;
   const {
     meta: {
-      auth: { requireAuth, guestOnly, requireClaims }
+      auth: { requireAuth, guestOnly, roles }
     }
   } = restrictedRoute;
 
-  if ((requireAuth || requireClaims) && !currentUser)
+  if ((requireAuth || roles) && !currentUser)
     return next({
       path: "/login",
       query: { redirect: to.fullPath }
     });
 
   if (guestOnly && currentUser) return next("/");
 
-  if (requireClaims) {
+  if (roles) {
     const userClaims = currentUser
-      ? (await currentUser.getIdTokenResult()).claims
+      ? (await currentUser.getIdTokenResult()).claims.roles
       : null;
 
-    return !userClaims || !hasClaim(requireClaims, userClaims)
+    return !userClaims || !store.getters["user/hasRole"](roles, userClaims)
```

Comment:
Can't we just delegate dealing with the claims to the store module altogether? Why getting claims here and passing them back to the sotre getter?